spec/web/schema/squad_requests_spec.exs

```diff
@@ -0,0 +1,114 @@
+defmodule Squadster.Web.Schema.SquadRequestSpec do
+  use ESpec.Phoenix, async: true
+  use ESpec.Phoenix.Extend, :controller
+  use Phoenix.ConnTest
+
+  import Squadster.Support.Factory
+
+  alias Squadster.Formations.SquadRequest
+
+  let :create do
+    """
+      mutation createSquadRequest($squad_id: String) {
+        createSquadRequest(squad_id: $squad_id) {
+          insertedAt
+        }
+      }
+    """
+  end
+
+  let :delete do
+    """
+      mutation deleteSquadRequest($id: Int) {
+        deleteSquadRequest(id: $id) {
+          insertedAt
+        }
+      }
+    """
+  end
+
+  let :approve do
+    """
+      mutation approveSquadRequest($id: Int) {
+        approveSquadRequest(id: $id) {
+          approvedAt
+        }
+      }
+    """
+  end
+
+  let :create_squad_request_query, do: %{query: create(), variables: %{squad_id: squad().id}}
+
+  let :squad, do: insert(:squad)
+  let :user, do: insert(:user)
+
+  def delete_squad_request_query(squad_id) do
+    %{query: delete(), variables: %{id: squad_id}}
+  end
+
+  def approve_squad_request_query(squad_id) do
+    %{query: approve(), variables: %{id: squad_id}}
+  end
+
+  def api_request(payload) do
+    login_as(user()) |> query(payload)
+  end
+
+  def entities_count(struct) do
+    struct |> Repo.all |> Enum.count
+  end
+
+  describe "mutations" do
+    context "create_squad_request" do
+      it "creates a new squad_request with valid attributes" do
+        previous_count = entities_count(SquadRequest)
+        api_request(create_squad_request_query())
+        expect entities_count(SquadRequest) |> to(eq previous_count + 1)
+      end
+
+      context "when user has another request" do
+        it "should delete old request and create new one" do
```

Comment:
I think we need more complex approach here, this test doesn't show us, what manipulations were in database and what happened with this requests. Spec's name doesn't match its implementation. If you don't agree, please, prove you opinion.

Reply:
Maybe, but I'm not sure how to test what's happening in the middle of the function, will lock at expect_receive now